executor/importer/job.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"context"
+	"encoding/json"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/parser/terror"
+	"github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/dbterror/exeerrors"
+	"github.com/pingcap/tidb/util/sqlexec"
+	"github.com/tikv/client-go/v2/util"
+)
+
+// vars used for test.
+var (
+	// TestLastImportJobID last created job id, used in unit test.
+	TestLastImportJobID atomic.Int64
+)
+
+// constants for job status and step.
+const (
+	// JobStatus
+	// 	┌───────┐     ┌───────┐    ┌────────┐
+	// 	│pending├────►│running├───►│finished│
```

Comment:
pending -> running -> reverting -> reverted/revert_failed

Reply:
that's frameworks's state transition, we don't have revert for job